internal/socket_test.go

```diff
@@ -24,3 +25,32 @@ func FuzzUcredToStringToUcred(f *testing.F) {
 		assert.Equal(t, ucred.Uid, resUcred.Uid)
 	})
 }
+
+func Test_authenticateUser(t *testing.T) {
+	tests := []struct {
+		name    string
+		uid     int
+		grps    []string
+		wantErr bool
+	}{
+		{
+			name:    "test1: existing user id 1000",
```

Comment:
Expecting in test environment to be available user with id 1000 and to belong to group `nordvpn` - now we have such in our tester docker container. Otherwise would need to setup groups and users which requires root access - dont have it in unit tests.

Reply:
We can assume that user running the tests can read the system users and find ID of the user matching the criteria